pkg/csi/cinder/openstack/openstack_volumes.go

```diff
@@ -312,5 +266,5 @@ func (os *OpenStack) diskIsUsed(volumeID string) (bool, error) {
 	if err != nil {
 		return false, err
 	}
-	return volume.AttachedServerId != "", nil
+	return volume.Attachments[0].ServerID != "", nil
```

Comment:
what if volume.Attachments is nil here ? may be you could check the same before.